bgpd: Show routes filtered by prefix-list in filter-routes command

adharkar opened the pull request:
Changed "show bgp ipv4 neighbor <peer> filtered-routes"
to show routes filtered by prefx lists, distribute lists and filter lists

Closes: #2653

Signed-off-by: Ameya Dharkar adharkar@vmware.com

Commit:
bgpd: Show routes filtered by prefix-list in filter-routes command

Changed "show bgp ipv4 neighbor <peer> filtered-routes"
to show routes filtered by prefx lists, distribute lists and filter lists

Closes: #2653

Signed-off-by: Ameya Dharkar adharkar@vmware.com

srimohans commented on bgpd/bgp_route.c:
```diff
@@ -10223,17 +10224,29 @@ static void show_adj_route(struct vty *vty, struct peer *peer, afi_t afi,
+					route_filtered = true;
+
+				/* Filter prefix using route-map */
+				if ((bgp_input_modifier(peer, &rn->p, &attr,
```
Use the route_filtered condition only to capture the result of bgp_input_filter(). Keep the RMAP_DENY condition as it.

srimohans commented on bgpd/bgp_route.c:
```diff
@@ -10223,17 +10224,29 @@ static void show_adj_route(struct vty *vty, struct peer *peer, afi_t afi,
 
 				if (type == bgp_show_adj_route_filtered
-				    && ret != RMAP_DENY) {
+				    && !route_filtered) {
```
Use the route_filtered condition only to capture the result of bgp_input_filter(). Keep the RMAP_DENY condition as it.

Commit:
bgpd: Show routes filtered by prefix-list in filter-routes command
Update:Addressed review comments

Changed "show bgp ipv4 neighbor filtered-routes"
to show routes filtered by prefx lists, distribute lists and filter lists

Closes: #2653

Signed-off-by: Ameya Dharkar adharkar@vmware.com
